torchgen/model.py

```diff
@@ -552,13 +568,19 @@ def from_yaml(
                         f"Dispatch key {dispatch_key} of kernel {v} "
                         "is not a supported dispatch key."
                     )
+                    # We only allow one level of namespace for kernels and operator.
+                    # We will append "native" to a custom kernel namespace.
+                    tokens = v.split("::", 1)
                     # Why is 'structured' included? External backends (e.g.
                     # XLA) opt into which ops are structured independently
                     # of which in-tree ops are structured
                     dispatch[dispatch_key] = BackendMetadata(
-                        v,
+                        kernel=tokens[-1],
                         structured=structured
                         and is_structured_dispatch_key(dispatch_key),
+                        cpp_namespace=f"{tokens[0]}::native"
+                        if len(tokens) > 1
+                        else "at::native",
```

Comment:
Tiny nit, but maybe it would be clearer that "`at` is the default namespace" if you write this as `namespace = tokens[0] if len(tokens) == 1 else 'at'; cpp_namespace = f"{namespace}::native"`, to make it clear what the default is?